File: fast_barcode_scanner/ios/Classes/Data/CameraInformation.swift
struct CameraInformation {
  let videoWidth: Int32
  let videoHeight: Int32

  let analysisWidth: Double
  let analysisHeight: Double

  init(videoWidth: Int32, videoHeight: Int32, analysisWidth: Int32, analysisHeight: Int32) {
    self.videoWidth = videoWidth
    self.videoHeight = videoHeight
    self.analysisWidth = Double(analysisWidth)
    self.analysisHeight = Double(analysisHeight)
  }
  var serialized: [String: Any] {
    let offsetX = Double(videoWidth) - Double(analysisWidth)
    let offsetY = Double(videoHeight) - Double(analysisHeight)

    return [
      "video_size": [Double(videoWidth), Double(videoHeight)],
      "analysis_frame": [offsetX / 2, offsetY / 2, Double(analysisWidth), Double(analysisHeight)],
    ]
  }
}

File: fast_barcode_scanner/android/src/main/kotlin/com/jhoogstraat/fast_barcode_scanner/types/CameraInformation.kt
package com.jhoogstraat.fast_barcode_scanner.types

data class CameraInformation(val videoWidth: Double,
                             val videoHeight: Double,
                             val analysisWidth: Double,
                             val analysisHeight: Double) {
    fun serialized() {
        val offsetX = videoWidth - analysisWidth
        val offsetY = videoHeight - analysisHeight

        hashMapOf(
            "video_size" to arrayOf(videoWidth, videoHeight),
            "analysis_frame" to arrayOf(offsetX / 2, offsetY / 2, analysisWidth, analysisHeight)
        )
    }
}
